ios/src/TangramMap/TGMapViewController.h

```diff
@@ -0,0 +1,731 @@
+- (void)mapView:(TGMapViewController *)view recognizer:(UIGestureRecognizer *)recognizer didRecognizeShoveGesture:(CGPoint)displacement;
+@end
+
+NS_ASSUME_NONNULL_END
+
+/**
+ A map view delegate that can be implemented to receive miscellaneous map event.
+
+ @note All of those methods are called from main thread, those methods are all **optional**.
+ */
+@protocol TGMapViewDelegate <NSObject>
+@optional
+/**
+ Called after a `-[TGMapViewController loadSceneFileAsync:]` or
+ `-[TGMapViewController loadSceneFileAsync:sceneUpdates:]` is completed.
+
+ @param mapView a pointer to the map view
+ @param scene the path to the scene that has been loaded
+ */
+- (void)mapView:(nonnull TGMapViewController *)mapView didLoadSceneAsync:(nonnull NSString *)scene;
+/**
+ Always called after the method `-[TGMapViewController pickFeatureAt:]` is called on the map view.
+
+ @param mapView a pointer to the map view
+ @param feature a dictionnary containing the data contained by the selected feature, may be `nil`
+ @param position the screen position at which the feature has been selected
+
+ @note If no feature have been selected, `feature` will be `nil`, and `position` `(0, 0)`.
+ */
+- (void)mapView:(nonnull TGMapViewController *)mapView didSelectFeature:(nullable NSDictionary *)feature atScreenPosition:(CGPoint)position;
+ /**
+ Always called after the method `-[TGMapViewController pickLabelAt:]` is called on the map view.
+
+ @param mapView a pointer to the map view
+ @param labelPickResult the label pick result object containing information about the label being picked, may be `nil`
+ @param position the screen position at which the feature has been selected
+
+ @note If no feature have been selected, `feature` will be `nil`, and `position` `(0, 0)`.
+ */
+- (void)mapView:(nonnull TGMapViewController *)mapView didSelectLabel:(nullable TGLabelPickResult *)labelPickResult atScreenPosition:(CGPoint)position;
+/**
+ Always called after the method `-[TGMapViewController pickMarkerAt:]` is called on the map view.
+
+ @param mapView a pointer to the map view
+ @param markerPickResult the marker pick result object containing information about the marker being picked, may be `nil`
+ @param position the screen position at which the feature has been selected
+
+ @note If no feature have been selected, `feature` will be `nil`, and `position` `(0, 0)`.
+ */
+- (void)mapView:(nonnull TGMapViewController *)mapView didSelectMarker:(nullable TGMarkerPickResult *)markerPickResult atScreenPosition:(CGPoint)position;
+/**
+ Called whenever the view did complete loading and building of all of the visible tiles of the current view.
+
+ @param mapView a pointer to the map view
+ */
+- (void)mapViewDidCompleteLoading:(nonnull TGMapViewController *)mapView;
+@end
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ `TGMapViewController` is a flexible and customizable map view, allowing you to embed
+ it inside a <a href="https://developer.apple.com/reference/glkit/glkview">GLKView</a> component.
```

Comment:
Mmmm TGMapViewController is a `GLKitViewController` which manages the lifecycle of a `GLKView` so this might need a little re-wording? In practice, you don't embed anything inside the `GLKView`, unless I'm misunderstanding something here?